src/stages/tx_lookup.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    common::hash_data,
+    etl::{
+        collector::{Collector, OPTIMAL_BUFFER_CAPACITY},
+        data_provider::Entry,
+    },
+    kv::tables,
+    models::BodyForStorage,
+    stagedsync::stage::{ExecOutput, Stage, StageInput},
+    txdb, MutableTransaction, StageId,
+};
+use async_trait::async_trait;
+use tokio::pin;
+use tokio_stream::StreamExt;
+use tracing::*;
+
+#[derive(Debug)]
+pub struct BlockHashes;
+
+#[async_trait]
+impl<'db, RwTx> Stage<'db, RwTx> for BlockHashes
+where
+    RwTx: MutableTransaction<'db>,
+{
+    fn id(&self) -> StageId {
+        StageId("TxLookup")
+    }
+
+    fn description(&self) -> &'static str {
+        "Generating TransactionHash => BlockNumber Mapping"
+    }
+
+    async fn execute<'tx>(&self, tx: &'tx mut RwTx, input: StageInput) -> anyhow::Result<ExecOutput>
+    where
+        'db: 'tx,
+    {
+        let past_progress = input.stage_progress.unwrap_or(0);
+
+        let mut bodies_cursor = tx.mutable_cursor(&tables::BlockBody).await?;
+        let mut tx_hash_cursor = tx.mutable_cursor(&tables::TxLookup).await?;
+
+        let start_key = past_progress.to_be_bytes();
+        let mut collector = Collector::new(OPTIMAL_BUFFER_CAPACITY);
+        let walker = txdb::walk(&mut bodies_cursor, &start_key, 0);
+        pin!(walker);
+
+        while let Some((block_key, ref body)) = walker.try_next().await? {
+            let (block_number, block_hash) = (&block_key[0..8], &block_key[8..]);
+
+            let body = rlp::decode::<BodyForStorage>(body)?;
+            for ref uncle in body.uncles {
```

Comment:
Why are you taking uncles, you need to take transactions

Reply:
Where are the txs?

Reply:
you need to use the transaction base id and transaction count from the block body you computed to extract them from the EthTx bucket/table/cursor

Reply:
I would suggest taking a read on the ReadTransactions function in Erigon